lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -116,7 +120,7 @@ def exit
         @stderr.close if !@stderr.closed?
 
         # wait up to 2 seconds for the watcher thread to fully exit
-        @ps_process.join(2)
+        @ps_process.join(2) if @ps_process.alive?
```

Comment:
Should we be iterating over `@@instances` and killing all instances not just this one?

Should there be a different method to `exit_all` ?